src/System.Linq.Expressions/src/System/Linq/Expressions/IndexExpression.cs

```diff
@@ -272,14 +272,13 @@ private static string GetArgTypesString(Expression[] arguments)
 
         private static PropertyInfo FindProperty(Type type, string propertyName, Expression[] arguments, BindingFlags flags)
         {
-            var props = type.GetProperties(flags).Where(x => x.Name.Equals(propertyName, StringComparison.CurrentCultureIgnoreCase)); ;
-            PropertyInfo[] members = new List<PropertyInfo>(props).ToArray();
-            if (members == null || members.Length == 0)
+            var members = type.GetProperties(flags)
+                .Where(x => x.Name.Equals(propertyName, StringComparison.CurrentCultureIgnoreCase));
+            if (!members.Any())
```

Comment:
Have you measured the perf throughput here?  Yes, your change removes the list/array allocation, which is good.  But you've also caused both the Any and the Where to allocate extra enumerators where they wouldn't have before, in order to enumerate in the Any() call, and this causes the Where to need to enumerate until either the end of the source or until the predicate returns true, whichever comes first.

Why is this !members.Any() call needed at all?  Without it, we'll simply call FindBestProperty, which will also do the enumeration (just the once), and if it's empty, FindBestProperty will return 0 and we'll end up returning null from the method anyway.